File: Eclipse/w7cpframe/src/clsPfCoordinate.java
//
// Copyright (c)2016 S C Harrison
// Refer to License.txt for terms and conditions of use.
//
import java.io.*;
import java.util.regex.*;


    class clsPfCoordinate
    {
        //Define Class

        public int key = 0;
        public double x = 0;           //.. x-coord of a joint ..
        public double y = 0;           //.. y-coord of a joint ..



        public void initialise()
        {
            key = 0;
            x = 0;
            y = 0;
        }

        public void setValues(int nodeKey, double x1, double y1)
        {
            key = nodeKey;
            x = x1;
            y = y1;
        }

        public String sprint()
        {
            String s;

            s = String.format("%4d %12.3f %12.3f%n",key,x,y);

            return s;
        }

        public void cprint()
        {
            System.out.print(sprint());
        }

        public void fprint(BufferedWriter fp)
        {
            try {
				fp.write(sprint());
			} catch (IOException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
        }

        public void fgetData(BufferedReader fp)
        {
            String s="";

            System.out.println("fgetData ...");

            try {
				s = fp.readLine();
			} catch (IOException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
            sgetData(s);

            System.out.println("... fgetData");
        }


        public void sgetData(String s)
        {
            String[] dataflds = new String[10]; //(0 To 9);
            int i;


            System.out.println("sgetData ...");

            //Trim Trailing Spaces from input String
            String regExpTrimPattern = "^\\s+|\\s+$"; //trim trailing spaces          
            Pattern regExpPatternObj = Pattern.compile(regExpTrimPattern);
            Matcher regExpMatches = regExpPatternObj.matcher(s);
            String regResult = regExpMatches.replaceAll("");
            
            //Get Data fields from input String
            String regExpNumberPattern = "-?\\d+(?:[,.]\\d+)?";
            regExpPatternObj = Pattern.compile(regExpNumberPattern);
            regExpMatches = regExpPatternObj.matcher(regResult);
            i=0;
            while (regExpMatches.find()){
            	dataflds[i] = regExpMatches.group().trim();
            	i = i+1;
            }
  
            //Assign data fields to named variables
            key = Integer.parseInt(dataflds[0]);
            x = Double.parseDouble(dataflds[1]);
            y = Double.parseDouble(dataflds[2]);

            System.out.println("... sgetData");
        }

    } // class

File: VisualStudio/VBasic/w7cpframe/clsPfCoordinate.vb
'
' Copyright (c)2016 S C Harrison
' Refer to License.txt for terms and conditions of use.
'
Option Explicit On

Public Class clsPfCoordinate

#Const isUsingExcel = 0

    Public key As Integer
    Public x As Double           '.. x-coord of a joint ..
    Public y As Double           '.. y-coord of a joint ..

    Sub initialise()
        key = 0
        x = 0
        y = 0
    End Sub

    Sub setValues(ByVal nodeKey As Integer, ByVal x1 As Double, ByVal y1 As Double)
        key = nodeKey
        x = x1
        y = y1
    End Sub

    Function sprint() As String
        sprint = StrLPad(Format(key, "###"), 8) & StrLPad(Format(x, "0.0000"), 12) & StrLPad(Format(y, "0.0000"), 12)
    End Function

    Sub cprint()
        Console.WriteLine(sprint)
    End Sub

    Sub fprint(ByVal fp As Integer)
        Print(fp, sprint)
    End Sub


#If isUsingExcel = 1 Then

    Sub wbkPrint(ByVal dataRec As Range, ByVal RecNum As Integer)
        With dataRec
            .Offset(RecNum, 0).Value = key
            .Offset(RecNum, 1).Value = x
            .Offset(RecNum, 2).Value = y
        End With
    End Sub

    Sub wbkRead(ByVal dataRec As Range, ByVal RecNum As Integer)
        With dataRec
            key = .Offset(RecNum, 0).Value
            x = .Offset(RecNum, 1).Value
            y = .Offset(RecNum, 2).Value
        End With
    End Sub

#End If

End Class
